### PR TITLE
StickyHeader | Review comments incorporated

### PR DESCRIPTION
**Pull request by Vipul6**

### CODE REVIEW

File: utils/util.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const useHandleScroll = (elemRef, setterFunc) => {
```
Comment (vinodloha) [inline utils/util.js]: The name of the hook is ambiguous and also not placed correctly. We are keeping hooks at lib/hooks/*

File: lib/index.js [hunk 2]
```diff
@@ -79,4 +80,6 @@ export {
   TypeaheadVanilla,
   Video,
   VideoVanilla,
+  StickyHeader,
```
Comment (vinodloha) [inline lib/index.js]: Follow the alphabetical order

File: lib/components/molecules/StickyHeader/StickyHeader.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  useEffect(() => {
+    window.addEventListener('scroll', () => useHandleScroll(wrapperRef, setSticky));
+    return () => {
+      window.removeEventListener('scroll', useHandleScroll);
+    };
+  });
+
+  return (
+    <div aria-label="Sticky Header" className={`${isSticky ? className : ''}`} ref={wrapperRef}>
+      <header>{children}</header>
```
Comment (vinodloha) [inline lib/components/molecules/StickyHeader/StickyHeader.js]: @Vipul6 I don't think you have fixed the original problem i had highlighted. This component is causing a flicker when it gets sticky and also hiding the page content below itself.

**Review by vinodloha — CHANGES_REQUESTED** [review]
This still needs few fixes

File: utils/util.js (hunk 1 above)
Comment (Vipul6) [reply utils/util.js]: It has been taken care.

File: lib/index.js (hunk 2 above)
Comment (Vipul6) [reply lib/index.js]: It has been taken care.

File: lib/components/molecules/StickyHeader/StickyHeader.js (hunk 3 above)
Comment (Vipul6) [reply lib/components/molecules/StickyHeader/StickyHeader.js]: Fixed.

File: lib/components/molecules/StickyHeader/StickyHeader.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  useEffect(() => {
+    window.addEventListener('scroll', () => useScroll(wrapperRef, setSticky));
+    return () => {
+      window.removeEventListener('scroll', useScroll);
+    };
+  });
+
+  return (
+    <Wrapper
+      aria-label="Sticky Header"
```
Comment (vinodloha) [inline lib/components/molecules/StickyHeader/StickyHeader.js]: Don't hard code any string in the component, take them through props please.

File: lib/components/molecules/StickyHeader/StickyHeader.story.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        </StickyHeader>
+        <p>
+          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has
+          been the industrys
+        </p>
+        <p>
+          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has
+          been the industrys
+        </p>
+        <StickyHeader>
```
Comment (vinodloha) [inline lib/components/molecules/StickyHeader/StickyHeader.story.js]: Tested this, still not working as expected. When this instance gets sticky it should not overlap the above header unless explicitly set to do so. You had to create a prop that takes sticky position value. 

We had discussed this in quite a detail.

File: lib/components/molecules/StickyHeader/StickyHeader.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      className={`${isSticky ? className : 'no-sticky'}`}
+      ref={wrapperRef}
+      elmRef={wrapperRef}
+    >
+      <header>{children}</header>
```
Comment (vinodloha) [inline lib/components/molecules/StickyHeader/StickyHeader.js]: If there could be multiple Header tag on page, it should be wrapped in Section

File: lib/components/molecules/StickyHeader/StickyHeader.style.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  header {
+    position: fixed;
+    top: 0;
+    width: 100vw;
+  }
+`;
+
+export const Wrapper = styled.div`
+  position: relative;
+  height: ${props => props.elmRef.current != null && `${props.elmRef.current.offsetHeight}px`};
```
Comment (vinodloha) [inline lib/components/molecules/StickyHeader/StickyHeader.style.js]: This should be set inline on the element.

File: lib/hooks/useScroll.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+const useScroll = (elemRef, setterFunc) => {
```
Comment (vinodloha) [inline lib/hooks/useScroll.js]: It should be throttled/debounced, else it will execute on each pixel

**Review by vinodloha — CHANGES_REQUESTED** [review]
Needs more improvement.

File: lib/components/molecules/StickyHeader/StickyHeader.js (hunk 4 above)
Comment (aashutosh-prakash) [reply lib/components/molecules/StickyHeader/StickyHeader.js]: aria-label wont work on non-interactive elements like Wrapper (originally a div tag).
Removed it from code.

**Comment by aashutosh-prakash** [comment]
Raised new PR with commit lint fixes.

**Pull request closed without merging**